### PR TITLE
Admin update category

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/75144

## WHAT (optional)
- Add feature admin update category.

## HOW
- Add feature admin update category.

## Evidence (Screenshot or Video)
Screenshot from 2024-05-08 10-06-25
Screenshot from 2024-05-08 10-06-12
Screenshot from 2024-05-08 10-01-10
Screenshot from 2024-05-08 10-01-03

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controllers/admin.category.controller.ts [hunk 1]
```diff
@@ -42,3 +47,41 @@ export const createCategory = [
...
+  asyncHandler(async (req: IAdminCategoryRequest, res: Response) => {    
+    const category = await categoryService.getCategoryByName(req.body.name);
+
+    if (category) {
+      res.send({
+        status: Status.FAIL,
+        message: getTranslatedMessage('error.categoryExist', req.query.lng),
+      });
+      return;
+    }
```
Comment (huongnt-2545) [inline src/controllers/admin.category.controller.ts]: phần check category bị trùng tên có xử lý cả ở action create và update -> e tách thành function not export để tránh lặp code nhé

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**